engine/src/main/scala/cromwell/engine/workflow/SingleWorkflowRunnerActor.scala

```diff
@@ -1,32 +1,41 @@
 package cromwell.engine.workflow
 
 import java.nio.file.Path
+import java.util.UUID
 
 import akka.actor.FSM.{CurrentState, Transition}
 import akka.actor._
+import akka.pattern.pipe
 import better.files._
-import cromwell.core.{JobOutput, WorkflowId, _}
-import cromwell.engine
+import cromwell.core.retry.SimpleExponentialBackoff
+import cromwell.core.{WorkflowId, ExecutionStore => _, _}
 import cromwell.engine._
-import cromwell.engine.workflow.OldStyleSingleWorkflowRunnerActor._
-import cromwell.engine.workflow.OldStyleWorkflowManagerActor.{WorkflowMetadata, _}
-import cromwell.engine.workflow.WorkflowActor.WorkflowSucceededState
-import cromwell.engine.workflow.WorkflowManagerActor.{SubmitWorkflowCommand, SubscribeToWorkflowCommand}
+import cromwell.engine.workflow.SingleWorkflowRunnerActor._
+import cromwell.engine.workflow.WorkflowManagerActor.SubmitWorkflowCommand
+import cromwell.services.MetadataServiceActor.{GetSingleWorkflowMetadataAction, GetStatus, WorkflowOutputs}
+import cromwell.services.ServiceRegistryClient
+import cromwell.util.PromiseActor._
 import cromwell.webservice.CromwellApiHandler._
-import cromwell.webservice.{WdlValueJsonFormatter, WorkflowMetadataResponse}
+import cromwell.webservice.PerRequest.RequestComplete
+import cromwell.webservice.metadata.MetadataBuilderActor
+import spray.http.StatusCodes
 import spray.json._
 
-import scala.util._
-@deprecated(message = "This class will not be part of the PBE universe", since = "May 2nd 2016")
-object OldStyleSingleWorkflowRunnerActor {
+import scala.concurrent.ExecutionContext.Implicits.global
+import scala.concurrent.duration._
+import scala.language.postfixOps
+import scala.util.{Failure, Try}
+
```

Comment:
I think it's worth putting a description in somewhere about what all is going on with the polling, why it's there, etc